tests/IntegrationTest/RegisterExternalStandardsTest.php

```diff
@@ -0,0 +1,191 @@
+     *                             This version is randomly selected from the PHPCS versions compatible
+     *                             with the PHP version used in the test.
+     *
+     * @return void
+     */
+    public function testRegisterOneStandardLocal($phpcsVersion)
+    {
+        $config                                             = $this->configOneStandard;
+        $config['require-dev']['squizlabs/php_codesniffer'] = $phpcsVersion;
+
+        $this->writeComposerJsonFile($config, static::$tempLocalPath);
+        $this->assertComposerValidates(static::$tempLocalPath);
+
+        // Install the dependencies.
+        $this->assertExecute(
+            sprintf('composer install --no-plugins --working-dir=%s', escapeshellarg(static::$tempLocalPath)),
+            0,    // Expected exit code.
+            null, // No stdout expectation.
+            null, // No stderr expectation.
+            'Failed to install dependencies.'
+        );
+
+        // Verify that the standard registers correctly.
+        $installCommand = sprintf(
+            'composer install-codestandards --no-ansi --working-dir=%s',
+            escapeshellarg(static::$tempLocalPath)
+        );
+        $installResult  = $this->executeCliCommand($installCommand);
+        $this->assertSame(0, $installResult['exitcode'], 'Exitcode for install-codestandards did not match 0');
+
+        $this->assertMatchesRegularExpression(
+            '`^PHP CodeSniffer Config installed_paths set to [^\s]+/dummy-subdir$`',
+            trim($installResult['stdout']),
+            'Installing the standards failed.'
+        );
+
+        // Make sure the CodeSniffer.conf file has been created.
+        $this->assertFileExists(
+            static::$tempLocalPath . '/vendor/squizlabs/php_codesniffer/CodeSniffer.conf'
+        );
+
+        // Verify that PHPCS sees the external standard.
+        $this->assertExecute(
+            '"vendor/bin/phpcs" -i',
+            0,                 // Expected exit code.
+            'and DummySubDir', // Expected stdout.
+            '',                // Empty stderr expectation.
+            'Running phpcs -i failed.',
+            static::$tempLocalPath
+        );
+
+        // Verify that PHPCS can with the external standard set as the standard.
+        $phpcsCommand = '"vendor/bin/phpcs" --standard=DummySubDir -e';
+        $phpcsResult  = $this->executeCliCommand($phpcsCommand, static::$tempLocalPath);
+
+        $this->assertSame(0, $phpcsResult['exitcode'], 'Exitcode for PHPCS explain did not match 0');
+        $this->assertMatchesRegularExpression(
+            '`DummySubDir \(1 sniffs?\)\s+[-]+\s+DummySubDir\.Demo\.Demo(?:[\r\n]+|$)`',
+            $phpcsResult['stdout'],
+            'Output of the PHPCS explain command did not match the expectation.'
+        );
+    }
+
+    /**
+     * Data provider.
+     *
+     * @return array
+     */
+    public function dataRegisterOneStandard()
```

Comment:
Keeping with the "Use a verb for method names", I'd suggest starting data provider methods with `provide` (similar to test methods starting with `test`).

Reply:
Ha, two different schools of conventions clashing here.

I'm used to test data providers always being prefixed with `data`.

I'm generally in the habit of naming data providers which are linked to one test the same as the test with `test` replaced by `data`.
When a data provider is used by multiple tests, a generic, descriptive name, but again prefixed with `data`, is used.

Reply:
I tend to name the providers `provide...` with a description of what they are providing, to make the purpose of the data more clear to those unfamiliar with the inner workings of the code. I falls in line with `test...`, `mock...` and `assert...`.

But yeah, if the worst thing I can find in the code is the naming of methods, the code itself is pretty good.

Reply:
Thanks ;-)

I generally make sure that the test methods which use data providers have a docblocks with full `@param` annotations, that the parameter names are descriptive and then have the data provider closely coupled by placing it directly following the test method and using named data providers, with each case having a useful name to know what's being tested + using the parameter names in the sub arrays as indexes as well.

Altogether it might be a bit over the top, but I would be very surprised if there could ever be confusion about the purpose of the data with a setup like that... 😇